flow-decomposition/src/main/java/com/powsybl/flow_decomposition/DecomposedFlow.java

```diff
@@ -26,6 +28,12 @@ public class DecomposedFlow {
     private final double xNodeFlow;
     private final double pstFlow;
     private final double internalFlow;
+    private final double acTerminal1Current;
+    private final double acTerminal2Current;
+    private final double nominalVoltageTerminal1;
+    private final double nominalVoltageTerminal2;
+    private final CurrentLimits currentLimitsTerminal1;
+    private final CurrentLimits currentLimitsTerminal2;
```

Comment:
Could you avoid adding these new fields to a `DecomposedFlow` ?

I do not understand why putting them in the `DecomposedFlow` object if you use them once and then never again as you can find those constant information in the branch.

I understand that the AC current might be relevant. 

From a green IT perspective, these nominal voltage and current limits fields are not mandatory for each rescaler.

A flow decomposition on a real networks (Core for example) could lead to a huge number of `DecomposedFlow` in memory. 

Instead of storing these information here, a better strategy might be to modify the API from`DecomposedFlowRescale.rescale(DecomposedFlow decomposedFlow)` to `DecomposedFlowRescale.rescale(DecomposedFlow decomposedFlow, Branch branch)`.

Thus, you will be able to find your required information (nominal voltage, current limit, ...) for each rescaler

Reply:
Thank you for your suggestions.
I refactored the DecomposedFlowRescaler API, please tell me what you think of it! @OpenSuze